src/simplify-calculation.js

```diff
@@ -0,0 +1,270 @@
+import {isCanonical} from "./utils";
+
+function groupBy(items, key) {
```

Comment:
Please add function-level descriptions.

Is this all original work or are there sources that should be cited here?

Reply:
Fixed, and rewrote to use a map so that it can handle properties with non-string values. 

I wrote the utility-function, if that's what you're asking?